extensions/shared/plan-upgrade-notification.js

```diff
@@ -15,9 +16,40 @@ if ( undefined !== typeof window && window.location ) {
 	const { query } = parseUrl( window.location.href, true );
 
 	if ( query.plan_upgraded ) {
-		dispatch( 'core/notices' ).createNotice(
-			'success',
-			__( 'Plan upgraded. You may use premium blocks now.', 'jetpack' )
-		);
+		const planUrl = `https://wordpress.com/plans/my-plan/${ window.location.host }`;
+
+		apiFetch( { path: '/jetpack/v4/site', parse: true } )
+			.then( response => {
+				const planName = response.data.plan.product_name;
```

Comment:
Oh, this is a bummer; I thought that `apiFetch` would deep parse and `data` would object at this point but it's not:

<img width="184" alt="Screenshot 2019-09-30 at 09 41 23" src="https://user-images.githubusercontent.com/87168/65854976-a66f6b80-e366-11e9-9b31-11971693a3eb.png">

Looks like you need to `JSON.parse( response.data )`after all. :-(

Good thing is that if parsing fails at this point, we'll just end up in `catch` block and it continues working without separate try-catch. :-)

Reply:
Nice catch (you can tell I was waiting for you to test this 😛 ). I'm gonna readd the manual parsing. :-)